File: delcafe/app/Models/Menu.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Menu extends Model
{
    use HasFactory;

    protected $table = 'menu';

    protected $fillable = [
        'id_admin',
        'nama_menu',
        'kategori',
        'harga',
        'stok',
        'gambar',
        'stok_terjual',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'harga' => 'integer',
        'stok' => 'integer',
        'stok_terjual' => 'integer',
    ];

    // Relasi dengan Admin
    public function admin()
    {
        return $this->belongsTo(Admin::class, 'id_admin', 'id');
    }

    // Relasi dengan DetailPemesanan
    public function detailPemesanans()
    {
        return $this->hasMany(DetailPemesanan::class, 'id_menu', 'id');
    }
}

File: Delbites/lib/models/menu.dart
class Menu {
  final int menuId;
  final int adminId;
  final String namaMenu;
  final String kategori;
  final double harga;
  final int stok;
  final String? gambar;

  Menu({
    required this.menuId,
    required this.adminId,
    required this.namaMenu,
    required this.kategori,
    required this.harga,
    required this.stok,
    this.gambar,
  });

  factory Menu.fromJson(Map<String, dynamic> json) {
    return Menu(
      menuId: json['menu_id'],
      adminId: json['admin_id'],
      namaMenu: json['nama_menu'],
      kategori: json['kategori'],
      harga: double.parse(json['harga'].toString()),
      stok: json['stok'],
      gambar: json['gambar'],
    );
  }
}
